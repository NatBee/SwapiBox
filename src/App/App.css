.App {
  text-align: center;
  background-color: #8E7842;
  width: 100%;
  height: 100%;
  position: absolute; }

.App-logo {
  height: 80px; }

.App-title {
  font-size: 1.5em; }

.App-intro {
  font-size: large; }

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg); }
  to {
    transform: rotate(360deg); } }
