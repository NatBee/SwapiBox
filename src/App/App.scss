.App {
  text-align: center;
  background-color: #8E7842;
  width: 100%;
  height: 100%;
  position: absolute
}

.img {
  display: flex;
  justify-content: space-between;
  padding: 30px 30px 0px;
}

.App-logo {
  height: 140px;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
