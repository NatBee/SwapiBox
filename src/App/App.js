import React, { Component } from 'react';
import Nav from '../Nav/Nav.js';
import ButtonContainer from '../ButtonContainer/ButtonContainer.js';
import ScrollContainer from '../ScrollContainer/ScrollContainer.js';
import CardContainer from '../CardContainer/CardContainer.js';
import DataCleaner from '../helper.js'
import './App.css';

class App extends Component {
  constructor() {
    super();
    this.state = {
      peopleData: [],
      planetData: [],
      vehicleData: [],
      movieData: [],
      source: '',
      favorites: []
    }

    this.cleaner = new DataCleaner()
  }

  componentDidMount() {
    this.getMovieData();
  }

  getMovieData = async () => {
    const randomNumber = Math.floor(Math.random() * Math.floor(6));
    const scrolls =  await this.cleaner.getMovieData();
    const title = scrolls.results[randomNumber].title
    const releaseDate = scrolls.results[randomNumber].release_date
    const openingCrawl = scrolls.results[randomNumber].opening_crawl
    const cleanData = {title, releaseDate, openingCrawl}

    this.setState ({
      movieData: [cleanData]
    })
  }

  handleClickPeople = async () => {
    if(this.state.peopleData.length === 0) {
      const people = await this.cleaner.getPeopleData();
      this.setState ({
        peopleData: people,
        source: 'people'
      });
    } else {
      this.setState ({
        source: 'people'
      });
    }
  }

  handleClickPlanets = async () => {
    if(this.state.planetData.length === 0) {
      const planet = await this.cleaner.getPlanetsData();
      this.setState ({
        planetData: planet,
        source: 'planets'
      });
    } else {
      this.setState ({
        source: 'planets'
      });
    }
  }

  handleClickVehicles = async () => {
    if(this.state.vehicleData.length === 0) {
      const vehicle = await this.cleaner.getVehiclesData();
      this.setState ({
        vehicleData: vehicle,
        source: 'vehicles'
      });
    } else {
      this.setState ({
        source: 'vehicles'
      });
    }
  }

  favorites = () => {
    console.log('hey there')
  }

  handleClick = (data) => {
    const favoritesArray = this.state.favorites;
    
    if(!favoritesArray.includes(data)) {
      this.setState ({
        favorites: [...favoritesArray, data]
      })
    } else {
      const removeFavorite = favoritesArray.filter(favorite => {
        return favorite !== data
      })
      this.setState ({
        favorites: removeFavorite
      })
    }
  } 

  render() {
    return (
      <div className="App">
        <Nav 
          className="App-header" 
          favorites={this.favorites}
        />
        <ScrollContainer className="scroll-container"/>
        <ButtonContainer 
          className="Btn-container" 
          handleClickPeople={this.handleClickPeople}
          handleClickPlanets={this.handleClickPlanets}
          handleClickVehicles={this.handleClickVehicles}
        />
        <CardContainer 
          peopleData={this.state.peopleData}
          planetData={this.state.planetData}
          vehicleData={this.state.vehicleData}
          source={this.state.source}
          handleClick={this.handleClick}
        />
      </div>
    );
  }
}

export default App;
