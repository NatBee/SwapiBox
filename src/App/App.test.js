import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme';
/* eslint-disable */ 

describe('App tests', () => {

  beforeEach(() => {
    renderedComponent = shallow(<App />)
  })

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should match the snapshot', () => {
    expect(renderedComponent).toMatchSnapshot();
  })

  it('should gather movie data for 1 movie', () => {
    expect(renderedComponent.state().movieData).length).toBe(1)
  })

  it('should gather clean data from people data set', () => {
    expect(renderedComponent.state().peopleData).length).toBe(10)
  })

  it('should gather clean data from planet data set', () => {
    expect(renderedComponent.state().planetData).length).toBe(10)
  })

  it('should gather clean data from vehicle data set', () => {
    expect(renderedComponent.state().vehicleData).length).toBe(10)
  })

  it('when People button is clicked, it should render 10 cards with .people className', () => {

  })

  it('when Planets button is clicked, it should render 10 cards with .planets className', () => {
    
  })

  it('when Vehicles button is clicked, it should render 10 cards with .vehicles className', () => {
    
  })

  it('when Favorites button is clicked, it should render the number of favorited cards', () => {
    
  })

})