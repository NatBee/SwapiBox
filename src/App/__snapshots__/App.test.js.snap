// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="logo"
          className="App-logo"
          src="Star-Wars-Logo.png"
/>,
        <h1
          className="App-title"
>
          SwapiBox
</h1>,
        <ButtonContainer
          className="Btn-container"
          favorites={Array []}
          handleClickFavorites={[Function]}
          handleClickPeople={[Function]}
          handleClickPlanets={[Function]}
          handleClickVehicles={[Function]}
/>,
        <CardContainer
          favorites={Array []}
          handleClick={[Function]}
          peopleData={Array []}
          planetData={Array []}
          source=""
          vehicleData={Array []}
/>,
        <ScrollContainer
          className="scroll-container"
          movieData={Array []}
/>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "logo",
          "className": "App-logo",
          "src": "Star-Wars-Logo.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "SwapiBox",
          "className": "App-title",
        },
        "ref": null,
        "rendered": "SwapiBox",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "Btn-container",
          "favorites": Array [],
          "handleClickFavorites": [Function],
          "handleClickPeople": [Function],
          "handleClickPlanets": [Function],
          "handleClickVehicles": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "favorites": Array [],
          "handleClick": [Function],
          "peopleData": Array [],
          "planetData": Array [],
          "source": "",
          "vehicleData": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "scroll-container",
          "movieData": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="logo"
            className="App-logo"
            src="Star-Wars-Logo.png"
/>,
          <h1
            className="App-title"
>
            SwapiBox
</h1>,
          <ButtonContainer
            className="Btn-container"
            favorites={Array []}
            handleClickFavorites={[Function]}
            handleClickPeople={[Function]}
            handleClickPlanets={[Function]}
            handleClickVehicles={[Function]}
/>,
          <CardContainer
            favorites={Array []}
            handleClick={[Function]}
            peopleData={Array []}
            planetData={Array []}
            source=""
            vehicleData={Array []}
/>,
          <ScrollContainer
            className="scroll-container"
            movieData={Array []}
/>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "logo",
            "className": "App-logo",
            "src": "Star-Wars-Logo.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "SwapiBox",
            "className": "App-title",
          },
          "ref": null,
          "rendered": "SwapiBox",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "Btn-container",
            "favorites": Array [],
            "handleClickFavorites": [Function],
            "handleClickPeople": [Function],
            "handleClickPlanets": [Function],
            "handleClickVehicles": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "favorites": Array [],
            "handleClick": [Function],
            "peopleData": Array [],
            "planetData": Array [],
            "source": "",
            "vehicleData": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "scroll-container",
            "movieData": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
